packages/flutter_tools/lib/src/build_system/build_system.dart

```diff
@@ -772,9 +772,12 @@ class Node {
       }
     }
 
-    // If we depend on a file that doesnt exist on disk, kill the build.
+    // If we depend on a file that doesnt exist on disk, mark the build as
+    // dirty. if the rule is not correctly specified, this will result in it
+    // always being rerun.
     if (missingInputs.isNotEmpty) {
-      throw MissingInputException(missingInputs, target.name);
+      _dirty = true;
+      invalidatedReasons.add(InvalidedReason.inputChanged);
```

Comment:
does `inputChanged` mean that a required file is missing?  The comment on this file says: `An input file has an updated hash.`.

Reply:
Added `InvalidatedReason.inputMissing` with a description